src/commands/dependencies.rs

```diff
@@ -0,0 +1,212 @@
+            if let Some(l) = latest {
+                client.download(&info.name, &VersionReq::STAR).await?;
+                let mut tree = TreeBuilder::new(format!(
+                    "{}:{}@{}",
+                    info.name.namespace(),
+                    info.name.name(),
+                    l.version
+                ));
+                if let ReleaseState::Released { content } = &l.state {
+                    let path = client.content().content_location(content);
+                    if let Some(p) = path {
+                        let bytes = fs::read(&p)?;
+                        let deps = parser.parse(&bytes)?;
+                        for dep in deps {
+                            let mut dep_parser = DependencyImportParser {
+                                next: dep.name.0,
+                                offset: 0,
+                            };
+                            let dep = dep_parser.parse()?;
+                            let version = dep.req.clone().to_string();
+                            let v = if version == "*" {
+                                "*".to_string()
+                            } else {
+                                format!("{{{}}}", version.replace(',', ""))
+                            };
+                            let child = tree.begin_child(format!("{}@{}", dep.name.to_string(), v));
+                            match dep.kind {
+                                ImportKind::Locked(_) | ImportKind::Unlocked => {
+                                    Self::parse_deps(
+                                        &PackageName::new(dep.name)?,
+                                        dep.req,
+                                        client,
+                                        child,
+                                        &mut parser,
+                                    )
+                                    .await?;
+                                }
+                                ImportKind::Interface(_) => {}
+                            }
+                            child.end_child();
+                        }
+                    }
+                    let built = tree.build();
+                    print_tree(&built)?
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+struct DepsParser {}
+
+impl DepsParser {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn parse_imports<'a>(
+        &mut self,
+        parser: ComponentImportSectionReader<'a>,
+        deps: &mut Vec<ComponentImport<'a>>,
+    ) -> Result<()> {
+        for import in parser.into_iter_with_offsets() {
+            let (_, imp) = import?;
+            deps.push(imp);
+        }
+        Ok(())
+    }
+
+    pub fn parse<'a>(&mut self, mut bytes: &'a [u8]) -> Result<Vec<ComponentImport<'a>>> {
+        let mut parser = Parser::new(0);
+        let mut deps = Vec::new();
+        loop {
+            let payload = match parser.parse(bytes, true)? {
```

Comment:
The primary obstacle I see here is that I lose the ability to do `parser.skip_section` when encountering `CodeSectionStart`, which seems to prevent wasi interface imports from making it into the tree being printed.  Can also spend some time looking for a way to make it work though.

Reply:
Makes sense :+1: